cpp/src/arrow/util/hashing.h

```diff
@@ -709,6 +730,10 @@ class BinaryMemoTable : public MemoTable {
     //
     // Thus, the method will properly inject an empty value of the proper width
     // in the output buffer.
+    //
+    if (start >= size()) {
```

Comment:
Could you add a comment for this new condition

Reply:
+1

Reply:
You want an API comment or internal documentation. This early return protect agaisn't buffer overrun, do you want me to use an assert instead?

Reply:
I guess the question: in which situation do we have `start >= size()`? The original code didn't protect against this.